zkevm-circuits/src/evm_circuit/util/math_gadget.rs

```diff
@@ -816,3 +816,248 @@ impl<F: FieldExt, const NUM_BYTES: usize> MaxGadget<F, NUM_BYTES> {
+    dividend: util::Word<F>,
+    divisor: util::Word<F>,
+    quotient: util::Word<F>,
+    remainder: util::Word<F>,
+    v0: [Cell<F>; 9],
+    lt_lo: LtGadget<F, 16>,
+    comparison_hi: ComparisonGadget<F, 16>,
+}
+
+impl<F: FieldExt> DivWordsGadget<F> {
+    pub(crate) fn construct(
+        cb: &mut ConstraintBuilder<F>,
+        dividend: util::Word<F>,
+        divisor: util::Word<F>,
+    ) -> Self {
+        let quotient = cb.query_word();
+        let remainder = cb.query_word();
+        let v0 = cb.query_bytes();
+
+        let mut dividend_limbs = vec![];
+        let mut divisor_limbs = vec![];
+        let mut quotient_limbs = vec![];
+        for idx in 0..4 {
+            let now_idx = (idx * 8) as usize;
+            dividend_limbs
+                .push(from_bytes::expr(&dividend.cells[now_idx..now_idx + 8]));
+            divisor_limbs
+                .push(from_bytes::expr(&divisor.cells[now_idx..now_idx + 8]));
+            quotient_limbs
+                .push(from_bytes::expr(&quotient.cells[now_idx..now_idx + 8]));
+        }
+        let remainder_lo = from_bytes::expr(&remainder.cells[0..16]);
+        let remainder_hi = from_bytes::expr(&remainder.cells[16..32]);
+
+        let t0 = divisor_limbs[0].clone() * quotient_limbs[0].clone();
+        let t1 = divisor_limbs[0].clone() * quotient_limbs[1].clone()
+            + divisor_limbs[1].clone() * quotient_limbs[0].clone();
+        let t2 = divisor_limbs[0].clone() * quotient_limbs[2].clone()
+            + divisor_limbs[1].clone() * quotient_limbs[1].clone()
+            + divisor_limbs[2].clone() * quotient_limbs[0].clone();
+        let t3 = divisor_limbs[0].clone() * quotient_limbs[3].clone()
+            + divisor_limbs[1].clone() * quotient_limbs[2].clone()
+            + divisor_limbs[2].clone() * quotient_limbs[1].clone()
+            + divisor_limbs[3].clone() * quotient_limbs[0].clone();
+
+        let cur_v0 = from_bytes::expr(&v0[..]);
+
+        let lt_lo = LtGadget::construct(
+            cb,
+            remainder_lo.clone(),
+            from_bytes::expr(&divisor.cells[0..16]),
+        );
+        let comparison_hi = ComparisonGadget::construct(
+            cb,
+            remainder_hi.clone(),
+            from_bytes::expr(&divisor.cells[16..32]),
+        );
+        let (lt_hi, eq_hi) = comparison_hi.expr();
+        let lt = select::expr(lt_hi, 1.expr(), eq_hi * lt_lo.expr());
+        cb.require_equal("remainder < quotient", 1.expr(), lt);
+
+        //radix_constant_64 == 2^64
+        //radix_constant_128 == 2^128
+        let radix_constant_64 = pow_of_two_expr(64);
+        let radix_constant_128 = pow_of_two_expr(128);
+        cb.require_equal(
+            "product(quotient, divisor)_lo + remainders_lo == dividends_lo + radix_lo ⋅ 2^128",
+            cur_v0.clone() * radix_constant_128,
+            t0.expr() + t1.expr() * radix_constant_64.clone()
+                + remainder_lo
+                - (dividend_limbs[0].clone()
```

Comment:
change it to `cb.require_equal("...", t0 + t1 * radix_constant_64 + remainders_lo, dividends_lo + cur_v0 * radix_constant_128)` to match the description.

btw. what's `radix_lo * 2^128` in the description? Isn't `radix_lo` equal to `2^128`?